### PR TITLE
docs: add description to explain upload process

### PR DESCRIPTION
**Pull request by lyjeileen**
## Change:
- Explain upload process in MultimodalInput's story

### Before
Screenshot 2024-05-04 at 8 53 50 PM

### After
Screenshot 2024-05-08 at 4 53 24 PM

### CODE REVIEW

File: src/components/input/multimodal/multimodalInput/multimodalInput.stories.tsx [hunk 1]
```diff
@@ -14,7 +14,7 @@ const meta: Meta<React.ComponentProps<typeof MultimodalInput>> = {
...
...imodalInput` component validates each file against criteria set by props, such as maximum file count, maximum file size, and accepted file types. Only files that meet all criteria are sent to the `uploadFileEndpoint`.\n\nThe `messageId` parameter is appended to the `uploadFileEndpoint` URL to link uploaded files to the corresponding message on the server. Upon successful upload, the server returns a `url` and `fileId` for each file. The `url` will be used in the file download functionality, while the `fileId` is used for potential file deletion.\n\nTo delete a file, the client sends the fileId and messageId to the deleteFileEndpoint. This action unlinks the file from the associated message and removes it from the server. If deletion is triggered before the file is fully uploaded, the uploading task is aborted to prevent incomplete uploads.\n\nThe `multimodalInput` component provides default error messages for uploading errors. However, you can augment these messages by passing additional information. Your custom error message will be appended to the default error message, providing more context for error resolution. Deleting errors do not accept customized error messages for now.',
```
Comment (kaseyvee) [inline src/components/input/multimodal/multimodalInput/multimodalInput.stories.tsx:17]:
```suggestion
          'The `MultimodalInput` component is a versatile form element that facilitates various types of user input. In addition to supporting text input, it empowers users to upload files seamlessly and efficiently. Designed to be flexible and adaptable, the `MultimodalInput` component serves as a foundation for accommodating diverse input requirements.\n\n**Explainaton of File Upload Process:**\n\nOnce the user selects files for upload, the `MultimodalInput` component validates each file against criteria set by props, such as maximum file count, maximum file size, and accepted file types. Only files that meet all criteria are sent to the `uploadFileEndpoint`.\n\nThe `messageId` parameter is appended to the `uploadFileEndpoint` URL to link uploaded files to the corresponding message on the server. Upon successful upload, the server returns a `url` and `fileId` for each file. The `url` will be used in the file download functionality, while the `fileId` is used for potential file deletion.\n\nTo delete a file, the client sends the fileId and messageId to the `deleteFileEndpoint`. This action unlinks the file from the associated message and removes it from the server. If deletion is triggered before the file is fully uploaded, the uploading task is aborted to prevent incomplete uploads.\n\nThe `MultimodalInput` component provides default error messages for uploading errors. However, you can augment these messages by passing additional information. Your custom error message will be appended to the default error message, providing more context for error resolution. Error messages that occur on deletion cannot be customized for now.',
```

Comment (kaseyvee) [reply src/components/input/multimodal/multimodalInput/multimodalInput.stories.tsx:17]: changed:
- `multimodalInput` > `MultimodalInput`
- deleteFileEndpoint > `deleteFileEndpoint`
- last sentence

**Review by kaseyvee — APPROVED** [review]

File: src/components/input/multimodal/multimodalInput/multimodalInput.tsx [hunk 2]
```diff
@@ -10,6 +10,16 @@ import type { FileData, Message, MultimodalInputProps } from '../../../types'
...
+/** The `MultimodalInput` component is a versatile form element that facilitates various types of user input. In addition to supporting text input, it empowers users to upload files seamlessly and efficiently. Designed to be flexible and adaptable, the `MultimodalInput` component serves as a foundation for accommodating diverse input requirements.
+
+__Explainaton of File Upload Process:__
+1. Client sends the files to server via REST APIs.
+2. Server transforms the files (or send them to cloud storage service for the transformation).
+3. Server saves the files to cloud storage service e.g. AWS S3.
+4. Server responds back to the client with urls.
+5. Client sends the links via WebSocket.
+6. WebSocket broadcasts the links to other users in the same chat.
+7. Client renders the links in the chat. */
```
Comment (Shiti) [inline src/components/input/multimodal/multimodalInput/multimodalInput.tsx:22]:
```suggestion
/** The `MultimodalInput` component  is designed to serve as a foundation for accommodating diverse input modes in conversational apps. 

__How does it work?__
Currently, it supports sending text as well as files. The text messages, are directly sent via the WebSocket while for files, it uses a combination of HTTP APIs and the WebSocket.

When user selects a file, it is sent to the registered HTTP API and that is used to indicate the upload progress for the file. The HTTP API is responsible for transforming and storing the file. 
Once user clicks send after adding text and/or files, the component sends the data over the WebSocket. For the files, it appends a list of the file names that were sent through the HTTP API earlier. The backend server of the app is expected to utilize the uploaded files along with the text when processing the message. 
```

Maybe you can rephrase it to make it a little less technical and easier to read. Maybe the last paragraph can be broken down in to steps.

Comment (lyjeileen) [reply src/components/input/multimodal/multimodalInput/multimodalInput.tsx:22]: How about this version? If it's not good enough, we can use your version
```suggestion
/** The `MultimodalInput` component  is designed to serve as a foundation for accommodating diverse input modes in conversational apps. Currently, it supports sending text as well as files. The text messages, are directly sent via the WebSocket while for files, it uses a combination of HTTP APIs and the WebSocket.
 
__How does file upload work?__

1. Users can select files to send alongside text messages.
2. Selected files are sent to a designated HTTP API.
3. The HTTP API can update the upload progress for each file and is responsible for transforming and storing the files securely.
5. Once uploaded, the file names are appended to the data sent over WebSocket.
6. The backend server of the application utilizes the uploaded files along with the text messages for message processing. */
```

Comment (Shiti) [reply src/components/input/multimodal/multimodalInput/multimodalInput.tsx:22]: I think either will work.

Comment (Shiti) [reply src/components/input/multimodal/multimodalInput/multimodalInput.tsx:22]: This section does not talk about removing/deleting the files before send

Comment (lyjeileen) [reply src/components/input/multimodal/multimodalInput/multimodalInput.tsx]: I added two sentences to describe file deletion

File: src/components/input/multimodal/multimodalInput/multimodalInput.tsx [hunk 3]
```diff
@@ -10,6 +10,16 @@ import type { FileData, Message, MultimodalInputProps } from '../../../types'
...
 import Uploader from '../uploader/uploader'
 
+/** The `MultimodalInput` component  is designed to serve as a foundation for accommodating diverse input modes in conversational apps. Currently, it supports sending text as well as files. The text messages, are directly sent via the WebSocket while for files, it uses a combination of HTTP APIs and the WebSocket.
+ 
+__How does file upload work?__
+
+1. Users can select files to send alongside text messages.
+2. Selected files are sent to a designated HTTP API.
+3. The HTTP API can update the upload progress for each file and is responsible for transforming and storing the files securely.
+4. If deletion is triggered before the file is fully uploaded, the uploading task is aborted using `AbortController`.
```
Comment (Shiti) [inline src/components/input/multimodal/multimodalInput/multimodalInput.tsx:20]: I think this is not needed here.
```suggestion
```

**Review by Shiti — APPROVED** [review]
